### PR TITLE
Domains: Update Domain Mapping checkout page

### PR DESCRIPTION
**Pull request by rafaelgallani**
#### Changes proposed in this Pull Request
This PR implements the changes on the domain mapping checkout page layout. 
Here's a comparison on how they looked before vs. how they're looking now: 

##### Before
before - desktop

##### After
Screen Shot 2021-08-19 at 22 16 35

##### Before
before - mobile

##### After
Screen Shot 2021-08-19 at 22 16 55

#### Testing instructions
 - Add a domain mapping to the cart (you will probably need to change the back end code to allow you to do so, otherwise this can only be done if your user is DWPO exempt);
 - Checkout and confirm that the new "thank you" page looks like this. 

_(If you already have a domain mapped and you have the receipt ID, you can bypass the checkout by going to `/checkout/thank-you/:your-site/:receipt-id` 😉)_

### CODE REVIEW

**Comment by rafaelgallani** [comment]
Huh. Looks like E2E tests are broken... Something related to the repo image, I have no idea why. 
I'll try force committing to trigger the tests again and see if that changes.

**Comment by AllTerrainDeveloper** [comment]
Yoast Thank you Page is still working fine with those changes.
image

As well as the Professional Email thank you page.
image

Changes look good to me :)

**Review by AllTerrainDeveloper — APPROVED** [review]

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Please fix the spelling here. I think it may also be a bit simpler to say just "Get professional email" and drop the "a".
```suggestion
							stepTitle: translate( 'Get professional email' ),
```

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Two questions here:
1. These look like the defaults - do we need to specify them?
2. If you do continue to pass them in, please translate the `title` and `description` strings.
```suggestion
			customSupportSection={ {
				description: translate( 'Have questions? Our Happiness Engineers are here if you need help.' ),
				links: [
					{
						href: CALYPSO_CONTACT,
						title: translate( 'Ask a question' ),
					},
					{
						href: SUPPORT_ROOT,
						title: translate( 'View support documentation' ),
					},
				],
			} }
```

File: client/components/thank-you/index.tsx
Comment (daledupreez) [inline client/components/thank-you/index.tsx]: Can you please change this back to `'View support documentation'`?
```suggestion
				title: translate( 'View support documentation' ),
```

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx:68]: Higher-level question: haven't we _already_ connected the domain? Or are more steps needed?

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Minor non-blocking note: it may be easier to follow the code if you use the `useTranslate` hook from `i18n-calypso` to access `translate` from within your component. I find that's easier to follow than it being injected via `localize`.
```suggestion
const DomainMappingThankYou = ( { domain } ) => {
	const translate = useTranslate();

	return (
```

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Two nits:
1. You don't have an `Internal dependencies` section and there are extra whitespace lines between the components
2. `@automattic/components` is technically external

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Both of the `stepCta` `Button` elements just have `href={ '#' }` and no `onClick` defined - I think you're going to want them linked to something.

**Review by daledupreez — COMMENTED** [review]
I dropped some comments on the PR.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (stephanethomas) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Do we really need to drop that `a`? `Get a professional email` read like a sentence, which looks fine to me.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: We could use `Add email` instead here, which is a CTA we already use on some pages of the email management section (and that also has the benefit of being already translated).

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Another nit: this should be `Set up your connection...`

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: For me this is mostly a subtle word usage thing: it doesn't feel right to be talking about getting "a professional email" because "a professional email" is a bit ambiguous between "a professional email service/address" and "a professional email in my inbox". "Get professional email" is probably still a little bit ambiguous, but I do feel it's better. 

Maybe "Add professional email" would be better all around, especially if we go with [your suggestion to use "Add email" for the CTA](https://github.com/Automattic/wp-calypso/pull/55561#discussion_r691109045). We could also use "Set up professional email", but I think that would imply that it already existed.

I can definitely see us making this a clearer upsell around the free trials, but that can wait for a later PR.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (rafaelgallani) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Forgot to call `translate` on them. Thanks!

File: client/components/thank-you/index.tsx
Comment (rafaelgallani) [reply client/components/thank-you/index.tsx]: Oops. I think I let this one pass when rebasing. Sorry! 😅

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (rafaelgallani) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx:68]: In this case, we still haven't. 
This is for when the user is purchasing a domain mapping - so it happens before the mapping itself. 
The mapping is right after that - by clicking the links which I [forgot to reference](https://github.com/Automattic/wp-calypso/pull/55561#discussion_r691077596) 😄

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (rafaelgallani) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: I relied too heavily on the linter... Not sure why that happened. Thanks 😄

**Comment by rafaelgallani** [comment]
@daledupreez @stephanethomas @AllTerrainDeveloper 
Thanks for the thorough review! 😄 
I'll address the suggestions and comments, but I would like a second opinion on the wording-related ones, so I'm pinging @poligilad-auto to have a look at them too.

**Review by poligilad-auto — CHANGES_REQUESTED** [review]
Regarding the wording:
I agree with @daledupreez we can go with "Add professional email" and a CTA of "Add email". 

And @rafaelgalani a few comments that can be applied to all the thank you pages:
1. The top blue area seems a bit big, can you remove the top and bottom padding that you added here? I'm referring to `padding: 50px 20px 30px;`
2. The buttons should be vertically aligned to the text on the left, currently it's top aligned. 
3. Title line height on mobile - can we decrease this? It looks too separated.
4. The padding on mobile is a bit large on the sides, can we change it to 24px?
5. Can we add more spacing between the "Go to setup" button and the content below on mobile? It can be 32px.
6. I see there is a grey div at the end of the page. Can we get rid of that and just have white?
7. Can we make the buttons full width in mobile view, and in that case also make the spacing be paragraph and the button below 16px instead of 12px?

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (delputnam) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx:68]: > Higher-level question: haven't we _already_ connected the domain? Or are more steps needed?

There may be a very small percentage of people who are mapping a domain that have already set the NS or A records prior purchasing a mapping, but that is very rare. We'll actually be adding in checks for this in a future sprint, so we can show a different message here for those cases.

**Comment by rafaelgallani** [comment]
@poligilad-auto 

- [X] The top blue area seems a bit big, can you remove the top and bottom padding that you added here? I'm referring to padding: 50px 20px 30px;

Removed it.

- [X] The buttons should be vertically aligned to the text on the left, currently it's top aligned.

Done. 😄 

- [X] Title line height on mobile - can we decrease this? It looks too separated.

Changed to `32px`. 
- [X] The padding on mobile is a bit large on the sides, can we change it to 24px?

Done. ✅ 

- [X] Can we make the buttons full width in mobile view, and in that case also make the spacing be paragraph and the button below 16px instead of 12px?

Done. ✅ 

- [X] I see there is a grey div at the end of the page. Can we get rid of that and just have white?

Done. ✅ 

---

- [ ] Can we add more spacing between the "Go to setup" button and the content below on mobile? It can be 32px.

The vertical spacing between the "Go to setup" button and the next section is `32px` already. Do you want to add more `32px` to it?

**Comment by rafaelgallani** [comment]
Here's how it's looking so far:





---

I'll update the PR screenshots once all the design changes are done.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Add Professional Email' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-mapping-checkout-page)

**Comment by poligilad-auto** [comment]
Hey @rafaelgalani,

> The vertical spacing between the "Go to setup" button and the next section is 32px already. Do you want to add more 32px to it?

No, it's fine like this. It looked different when I tested it.

> Title line height on mobile - can we decrease this? It looks too separated.
> Changed to 32px.

I'm not seeing that for some reason.. And please change the line height to 40px
image

Now the spacing on the bottom looks too small on mobile. Is it the same like the spacing they are using [here](https://github.com/Automattic/wp-calypso/pull/55311#issuecomment-900388351)?

image

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: Can you use `emailManagementPurchaseNewEmailAccount` here instead? We split out the intentional "purchase new email" content into the new path.

File: client/my-sites/checkout/checkout-thank-you/domains/style.scss
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/style.scss:11]: Does it make more sense to move this chunk of CSS into the core thank you component styles? We're going to want this across all of the post-checkout thank you pages, not just the domains one.

File: client/my-sites/checkout/checkout-thank-you/domains/style.scss
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/style.scss]: How many of these styles are actually more general things that we should be fixing for all of the post-checkout pages? I am sure we have a bunch of issues with the CSS due to us targeting a more "normal" Calypso context/frame, but all of these CSS tweaks feel like they should most likely be moved into the main ThankYou CSS, probably behind the `.is-section-checkout-thank-you` top-level selector.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: These are still missing `translate` calls and still duplicate the default values.

Might it be simpler to just remove this prop definition from here altogether?
```suggestion
```

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: I think this `alt` text should be translated.
```suggestion
			thankYouImage={ { alt: translate( 'Domain Connected' ), src: domainConnectedSuccess } }
```

**Review by daledupreez — COMMENTED** [review]
Thanks for taking all the feedback onboard, @rafaelgalani!

My main concerns after your updates are that a lot of these style updates/fixes are only being applied to the domains thank you content. Can we push more of those changes into either the checkout-specific or component-specific CSS files? I really want all these improvements to be done somewhere central so we can leverage them as we move other thank you pages across to the new designs.

**Comment by rafaelgallani** [comment]
> My main concerns after your updates are that a lot of these style updates/fixes are only being applied to the domains thank you content. Can we push more of those changes into either the checkout-specific or component-specific CSS files? I really want all these improvements to be done somewhere central so we can leverage them as we move other thank you pages across to the new designs.

@daledupreez In this PR we are just focusing on the domain flows to make the review cycle easier and to ensure that there are no unexpected impacts on other flows.
Whenever you’re ready to re-use these styles for other flows, you can move all the styling changes under `.checkout-thank-you__domains` and its nested selectors into the main `ThankYou` component styles. But we’d really like for that to be done in a separate PR.

File: client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx
Comment (rafaelgallani) [reply client/my-sites/checkout/checkout-thank-you/domains/domain-mapping-thank-you.jsx]: I was waiting for Poli's answer on the `customSupportSection` prop. But yes, that will be removed 😄 
And thanks for pointing out this missing `translate`.

**Comment by daledupreez** [comment]
> @daledupreez In this PR we are just focusing on the domain flows to make the review cycle easier and to ensure that there are no unexpected impacts on other flows.
Whenever you’re ready to re-use these styles for other flows, you can move all the styling changes under .checkout-thank-you__domains and its nested selectors into the main ThankYou component styles. But we’d really like for that to be done in a separate PR.

Thanks for clarifying your intended scope, @rafaelgalani.

The one gotcha is that you're the first team folding this into actual checkout, so there aren't any production flows to be worried about! 🙃 The only production use case for the new component is outside of checkout as part of a Professional Email provisioning flow.

I've been pushy here mostly because I am shepherding some of these design improvements along, and we have three teams likely to work with the component in the near future, so I want to make sure we keep our chances for repeated work to a minimum. It's often easier for someone in the code making CSS changes to have context as to whether they're are general or not, and it's often more time-consuming for another team to step in, get context, and then refactor those changes later. So I have pushed for doing them earlier, because I am honestly not sure when my team will be able to take this on!

**Comment by delputnam** [comment]
> The one gotcha is that you're the first team folding this into actual checkout, so there aren't any production flows to be worried about! 🙃 The only production use case for the new component is outside of checkout as part of a Professional Email provisioning flow.
> 
> I've been pushy here mostly because I am shepherding some of these design improvements along, and we have three teams likely to work with the component in the near future, so I want to make sure we keep our chances for repeated work to a minimum. It's often easier for someone in the code making CSS changes to have context as to whether they're are general or not, and it's often more time-consuming for another team to step in, get context, and then refactor those changes later. So I have pushed for doing them earlier, because I am honestly not sure when my team will be able to take this on!

@daledupreez I know you want to try to keep repeat work to a minimum, but I think that in this case since we're all working on these components at the same time and Nomado is focused specifically on the domains flow, I'd like to try to avoid having to ensure that any design changes that we make here will work for all of the teams and getting into a longer review cycle.

Once the frequency of updates settle down a bit, I think this page can be revisited to try to simplify and ensure things are DRY. I know we don't know when that will be , but I'd really prefer to keep the changes in this PR laser focused on the domains flow for now.

And now that we all know that this component will likely require more for future changes, we can make better plans for coordinating on things like this in the future.

**Comment by daledupreez** [comment]
💯 @delputnam! I was hoping to provide context for my team jumping all over the PR!

@rafaelgalani calling that out of scope was completely fair. I am honestly sorry if my explanation came across as a push to change the scope -- that was not my intent.

**Comment by rafaelgallani** [comment]
@daledupreez I understand, it's fine! Again, thanks for the (re) review! 
>I am honestly sorry if my explanation came across as a push to change the scope -- that was not my intent.

It didn't, I just wanted to make this PR as simple as possible. We can do look for it in a further PR 😄

**Comment by rafaelgallani** [comment]
> I'm not seeing that for some reason.. And please change the line height to 40px

@poligilad-auto I misunderstood your comment. Sorry for that. 

> Now the spacing on the bottom looks too small on mobile. Is it the same like the spacing they are using [here](https://github.com/Automattic/wp-calypso/pull/55311#issuecomment-900388351)?

It is, but on their examples, there's not a mobile viewport screenshot with a subtitle - only with a title. I will add a little more spacing on the mobile viewport 😸 

Here's how it's looking:

_(Please ignore the black background. That is a local chrome extension 😅)_
 
Screen Shot 2021-08-19 at 16 25 46

Can you take a look at it again? 😄

**Review by poligilad-auto — APPROVED** [review]
Looking good @rafaelgalani 👌

**Comment by rafaelgallani** [comment]
Needed to rebase on `trunk`.

**Review by delputnam — APPROVED** [review]
Approving on behalf of Nomado. Thanks for all the hard work on this one @rafaelgalani!

**Review by daledupreez — APPROVED** [review]
I haven't tested, but I think we've had all of Letero's comments addressed, so please feel free to move forward with getting this PR shipped. 🚀

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6455352

Thank you @rafaelgalani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6455352).

**Pull request merged**
